tldw_Server_API/app/core/MCP_unified/modules/implementations/flashcards_module.py

```
@@ -647,6 +666,15 @@ def _update_card_sync(self, context: Any, args: dict[str, Any]) -> dict[str, Any
             updates = args.get("updates", {})
             expected_version = args.get("expected_version")
             tags = args.get("tags")
+            workspace_id = self._workspace_id_from_context(context)
+
+            card = db.get_flashcard(card_uuid)
+            self._assert_card_workspace(card, workspace_id)
+
+            if "deck_id" in updates and workspace_id is not None:
+                destination_deck = db.get_deck(updates["deck_id"])
+                if destination_deck is None or str(destination_deck.get("workspace_id") or "").strip() != workspace_id:
+                    raise PermissionError("Flashcard access denied for workspace")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)